BonusCalcApi.Tests/V1/UseCase/UpdateReportSentAtUseCaseTests.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Threading.Tasks;
+using AutoFixture;
+using BonusCalcApi.Tests.V1.Helpers;
+using BonusCalcApi.V1.Exceptions;
+using BonusCalcApi.V1.Factories;
+using BonusCalcApi.V1.Gateways.Interfaces;
+using BonusCalcApi.V1.Infrastructure;
+using BonusCalcApi.V1.UseCase;
+using FluentAssertions;
+using FluentAssertions.Common;
+using Moq;
+using NUnit.Framework;
+
+namespace BonusCalcApi.Tests.V1.UseCase
+{
+    public class UpdateReportSentAtUseCaseTests
+    {
+        private UpdateReportSentAtUseCase _classUnderTest;
+        private Fixture _fixture;
+        private Mock<ITimesheetGateway> _timesheetGatewayMock;
+
+        [SetUp]
+        public void Setup()
+        {
+            _fixture = FixtureHelpers.Fixture;
+
+            _timesheetGatewayMock = new Mock<ITimesheetGateway>();
+
+            _classUnderTest = new UpdateReportSentAtUseCase(_timesheetGatewayMock.Object, InMemoryDb.DbSaver);
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            InMemoryDb.Teardown();
+        }
+
+        [Test]
+        public async Task ReportSentAtIsUpdateWhenNull()
+        {
+            // Arrange
+            var now = DateTime.UtcNow;
+            var timesheet = CreateExistingTimesheet(null);
+
+            // Act
+            await _classUnderTest.ExecuteAsync(timesheet.OperativeId, timesheet.WeekId);
+
+            // Assert
+            timesheet.ReportSentAt.Should().BeOnOrAfter(now);
+            InMemoryDb.DbSaver.VerifySaveCalled();
+        }
+
+        [Test]
+        public async Task ReportSentAtIsNotUpdatedWhenNotNull()
```

Comment:
What about if an operative's report needs sending twice?

Reply:
It's not to prevent sending another report but for recording whether a report has been sent or not so that if the close week process is interrupted it can resume without some operatives having two copies delivered. There's currently not a way in the wireframes to resend a report that has already been sent - the supervisor/manager would download the report via the operative page.